### ISSUE TITLE
"Unable to parse git host URL" error

### ISSUE THREAD

**Issue by mitar** [vweevers/hallmark#90]
When running inside a git clone which is using SSH git location.

**Comment by mitar** [comment]
Version 4.0.0. The full error is:

```
Error: Unable to parse git host URL "git@gitlab.com:tozd/go/errors.git"
```

It looks like it might be connected with the fact that I use sub-group on GitLab?

**Comment by vweevers** [comment]
I'm not familiar with GitLab sub-groups, but judging by that URL, I'd guess so, yes. Note that `hallmark` is currently geared towards GitHub so you may run into other issues too (just a heads up).

**Comment by vweevers** [comment]
Support of sub-groups is blocked on https://github.com/npm/hosted-git-info/issues/54 which is used internally here.

**Comment by dbrakman** [comment]
Addressing OP's use case seems to work with hosted-git-info 6.1.1:
```js
hostedGitInfo = requires("hosted-git-info")
hostedGitInfo.fromUrl("git@gitlab.com:tozd/go/errors.git")
```

I was able to use hallmark to autogenerate a draft changelog at my workplace for a repo that has a remote url like `git@gitlab.com:company/department/team/subteam/project-collection/project.git` with the following addition to a local installation of hallmark's package.json:
```js
  "overrides": {
    "hosted-git-info": "6.1.1"
  },
```

It's a bit scattered, but it seems that gitlab subgroups were marked as supported upstream in [hosted-git-info #56](https://github.com/npm/hosted-git-info/issues/56) and [hosted-git-info #43](https://github.com/npm/hosted-git-info/issues/43#event-2532028441) (though this commit is supposedly in version 3.0.8 with which I also had this issue)

The use case in the [hosted-git-info #54](https://github.com/npm/hosted-git-info/issues/54) issue linked as a blocker of this one has the overriding problem of self-hosted gitlab, which has been claimed in [hosted-git-info #23](https://github.com/npm/hosted-git-info/issues/23#issuecomment-287494014) and [#11](https://github.com/npm/hosted-git-info/issues/11#issuecomment-287495178)to be out of scope for "hosted git info" as URL conventions for self-hosted version control systems (VCS) are not necessarily discernible from the domain.

It seems like the [dreaded](https://common-changelog.org/#42-conventional-commits) Conventional Commits tool [tried](https://github.com/conventional-changelog/conventional-changelog/commit/514c4e2a5d9abe3e3a728170444f880f2f5984c9) and [failed](https://github.com/conventional-changelog/conventional-changelog/pull/187#event-1800966342) to fall back to provided domains, but I like their moxie. If the possibility of generating broken VCS links could be nonfatal or skipped, enabling the autogeneration of draft changelogs for self-hosted enterprise VCS users could greatly broaden the scope of adoption of this standard. It'd still be a draft requiring human review, and the target audience likely has enough familiarity with regexes or find-and-replace to fix any broken links.

<details><summary>I'm pretty new to node, so for my benefit as much as anyone else's, the commands specifically used to demonstrate success are within:</summary>
<p>

```bash
export PATH_TO_LOCAL_HALLMARK='/path/to/hallmark'
export PATH_TO_MY_REPO='/path/to/myrepo'
git clone https://github.com/vweevers/hallmark.git $PATH_TO_LOCAL_HALLMARK
cd $PATH_TO_LOCAL_HALLMARK
printf '{
  "override": {
    "hosted-git-info": "6.1.1"
  },' > tmpfile
awk 'NR > 1 {print}' package.json >> tmpfile
mv tmpfile package.json
npm install --save-dev .
npm link
cd $PATH_TO_MY_REPO
chmod +x ~/homebrew/lib/node_modules/hallmark/compat/cli.cjs
hallmark cc init
```

Observe new, nonempty file at `$PATH_TO_MY_REPO/CHANGELOG.md`.

</p>
</details>

**Comment by koppor** [comment]
I am using GitHub and get the same error:

```cmd
> hallmark lint CHANGELOG.md
Error: Unable to find git host
    at parse (C:\Users\koppor\AppData\Roaming\npm\node_modules\hallmark\node_modules\find-githost\index.js:129:11)
    at exports.fromGit (C:\Users\koppor\AppData\Roaming\npm\node_modules\hallmark\node_modules\find-githost\index.js:110:10)
    at repo (file:///C:/Users/koppor/AppData/Roaming/npm/node_modules/hallmark/index.js:224:13)
    at hallmark (file:///C:/Users/koppor/AppData/Roaming/npm/node_modules/hallmark/index.js:29:22)
    at Module.lint (file:///C:/Users/koppor/AppData/Roaming/npm/node_modules/hallmark/index.js:153:10)
    at file:///C:/Users/koppor/AppData/Roaming/npm/node_modules/hallmark/cli.js:33:14
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
```

The issue is when using `git worktree`, when `.git` is a file and not I directory.

```bash
$ cat .git
gitdir: C:/git-repositories/jabref-all/jabref/.git/worktrees/jabref
```

**Comment by vweevers** [comment]
@koppor That's a separate issue. A PR is welcome over at https://github.com/vweevers/find-gitdir (which does support git-style symlinks, for support of submodules, but it may need something more to also support worktrees).

**Comment by imi-kitten** [comment]
yeah, i am hitting this same error, i just want this tool for changelog linting, i don't need any fixing, and using this or the remarkable-cli and plugin itself results in the same error,

**Comment by vweevers** [comment]
I'll see if I can update `hosted-git-info` which at least should bring support of subgroups. Updating that dependency is long overdue.

**Comment by vweevers** [comment]
Fixed in [5.0.2](https://github.com/vweevers/hallmark/releases/tag/v5.0.2)
